packages/api/src/command/medical/hie/get-hie-initiator.ts

```diff
@@ -0,0 +1,54 @@
+import { Patient } from "@metriport/core/domain/patient";
+import { MetriportError } from "@metriport/core/util/error/metriport-error";
+import { isOboFacility } from "../../../domain/medical/facility";
+import { getPatientWithDependencies } from "../patient/get-patient";
+
+export type HieInitiator = {
+  oid: string;
+  name: string;
+  npi: string;
+  facilityId: string;
+};
+
+export async function getHieInitiator(
+  patient: Pick<Patient, "id" | "cxId">,
+  facilityId?: string
+): Promise<HieInitiator> {
+  const { organization, facilities } = await getPatientWithDependencies(patient);
+  if (!facilityId && facilities.length > 1) {
+    throw new MetriportError(
+      `Patient has more than one facility, facilityId is required`,
+      undefined,
+      {
+        patientId: patient.id,
+        facilities: facilities.length,
+      }
+    );
+  }
+  const facility = facilityId ? facilities.find(f => f.id === facilityId) : facilities[0];
+  if (!facility) {
+    if (facilityId) {
+      throw new MetriportError(`Patient not associated with given facility`, undefined, {
+        patientId: patient.id,
+        facilityId,
+      });
+    }
+    throw new MetriportError(`Could not determine facility for patient`, undefined, {
+      patientId: patient.id,
+    });
+  }
+  if (isOboFacility(facility.type)) {
```

Comment:
Turns out this check isn't good enough in this function. 
For CQ, I created this function: 
```
export function isCqOboFacility(facility: Facility): boolean {
  if (isOboFacility(facility.type)) {
    return facility.cqOboActive && facility.cqOboOid != null;
  }
  return false;
}
```

I think we need a similar one for CW. 

Then, we should probably pass one of `isCqOboFacility` or `isCwOboFacility` functions as a parameter to this function, so it can perform the corresponding check on line 40.